### ISSUE TITLE
Server error

### ISSUE THREAD

**Issue by MedpejGit** [FourLeafTec/tftp#2]
Hi, I implemented the tftp server example and I'm having the following error:

E/flutter ( 3827): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Unsupported operation: Cannot add to a fixed-length list
E/flutter ( 3827): #0      FixedLengthListMixin.add (dart:_internal/list.dart:21:5)
E/flutter ( 3827): #1      TFtpServerSocket._readFileNameAndTransType (package:tftp/src/server.dart:220:23)
E/flutter ( 3827): #2      TFtpServerSocket.read (package:tftp/src/server.dart:106:20)
E/flutter ( 3827): #3      TFtpServer._init.<anonymous closure> (package:tftp/src/server.dart:45:26)
E/flutter ( 3827): #4      _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
E/flutter ( 3827): #5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
E/flutter ( 3827): #6      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter ( 3827): #7      _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:784:19)
E/flutter ( 3827): #8      _StreamController._add (dart:async/stream_controller.dart:658:7)
E/flutter ( 3827): #9      _StreamController.add (dart:async/stream_controller.dart:606:5)
E/flutter ( 3827): #10     new _RawDatagramSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:2535:33)
E/flutter ( 3827): #11     _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:1379:14)
E/flutter ( 3827): #12     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
E/flutter ( 3827): #13     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)

Can you help me ?

Thank you!

**Comment by FourLeafTec** [comment]
If it's not too much trouble, could you please provide me with the output of the `flutter doctor -v` command?

The reason for the error is most likely that I haven't updated the SDK in a long time, and some of the APIs have changed. I will update the SDK to 3.3.3 soon.

**Comment by FourLeafTec** [comment]
Here is an untested commit. If you're in a hurry, you can check it out and try it out first.

[fa3df23](https://github.com/FourLeafTec/tftp/commit/fa3df236e4e0761186c2c7b5774cd71c11b81f7a)

```yaml
  tftp:
    git:
      url: https://github.com/FourLeafTec/tftp.git
      ref: fa3df23 # commit hash
```

**Comment by FourLeafTec** [comment]
fixed by 1.2.0
